ipa-core/src/protocol/context/dzkp_validator.rs

```diff
@@ -165,36 +165,86 @@ impl<'a> Segment<'a> {
         }
     }
 
+    /// This function returns the length of the first entry, i.e. `x_left`
+    ///
+    /// The lengths of the entries might be inconsistent, for asserting that the length is identical
+    /// across entries, call function `assert_len`
     #[must_use]
     pub fn len(&self) -> usize {
         self.x_left.len()
     }
 
+    /// This function allows to assert that the length of all entries is identical
+    ///
+    /// This function is mainly for debugging purposes but it can be used together with `len` to
+    /// determine that the segment is fully specified, i.e. all entries are not empty.
+    #[must_use]
+    pub fn assert_len(&self) -> bool {
+        (self.x_left.len() == self.x_right.len())
+            & (self.x_left.len() == self.y_left.len())
+            & (self.x_left.len() == self.y_right.len())
+            & (self.x_left.len() == self.prss_left.len())
+            & (self.x_left.len() == self.prss_right.len())
+            & (self.x_left.len() == self.z_right.len())
+    }
+
+    /// This function checks whether all entries are empty
+    ///
+    /// if one of the entries is not empty, the function returns `false`
     #[must_use]
     pub fn is_empty(&self) -> bool {
         self.x_left.is_empty()
+            & self.x_right.is_empty()
+            & self.y_left.is_empty()
+            & self.y_right.is_empty()
+            & self.prss_left.is_empty()
+            & self.prss_right.is_empty()
+            & self.z_right.is_empty()
+    }
+
+    /// This function allows to set the `SegmentEntries` `x_left` and `x_right` to the `Segment`
+    pub fn set_x(&mut self, x_left: SegmentEntry<'a>, x_right: SegmentEntry<'a>) {
+        self.x_left = x_left;
+        self.x_right = x_right;
+    }
+
+    /// This function allows to set the `SegmentEntries` `y_left` and `y_right` to the `Segment`
+    pub fn set_y(&mut self, y_left: SegmentEntry<'a>, y_right: SegmentEntry<'a>) {
+        self.y_left = y_left;
+        self.y_right = y_right;
+    }
+
+    /// This function allows to set the `SegmentEntries` `prss_left` and `prss_right` to the `Segment`
+    pub fn set_prss(&mut self, prss_left: SegmentEntry<'a>, prss_right: SegmentEntry<'a>) {
+        self.prss_left = prss_left;
+        self.prss_right = prss_right;
+    }
+
+    /// This function allows to set the `SegmentEntry` `z_right` to the `Segment`
+    pub fn set_z(&mut self, z_right: SegmentEntry<'a>) {
+        self.z_right = z_right;
     }
 }
 
 /// `SegmentEntry` is a simple wrapper to represent one entry of a `Segment`
 /// currently, we only support `BitSlices`
-#[derive(Clone, Debug)]
-pub struct SegmentEntry<'a>(&'a BitSliceType);
+#[derive(Clone, Debug, Default)]
+pub struct SegmentEntry<'a>(Option<&'a BitSliceType>);
```

Comment:
Not sure if the incremental construction will ultimately be necessary (as mentioned in another comment), but if it is, the builder idiom might be useful here. That would work like this: define a `SegmentEntryBuilder` with `Option`s like `SegmentEntry` has here. Implement a method `SegmentEntryBuilder::build(self) -> SegmentEntry` that confirms all of the options are populated (and does the other applicable sanity checks).